File: benchmarks/sieve.c
#include <stdlib.h>
#include <stdint.h>
#include <string.h>

int main(void) {
    int N = 1000000;
    uint8_t *sieve = (uint8_t *)malloc(N);
    memset(sieve, 1, N);
    for (int p = 2; (long)p * p < N; p++) {
        if (sieve[p]) {
            for (int j = p * p; j < N; j += p)
                sieve[j] = 0;
        }
    }
    int count = 0;
    for (int i = 2; i < N; i++)
        if (sieve[i]) count++;
    return count & 0xFF;
}

File: benchmarks/sieve.rs
fn main() {
    let n: usize = 1000000;
    let mut sieve = vec![true; n];
    let mut p = 2;
    while p * p < n {
        if sieve[p] {
            let mut j = p * p;
            while j < n {
                sieve[j] = false;
                j += p;
            }
        }
        p += 1;
    }
    let count = sieve[2..].iter().filter(|&&x| x).count();
    std::process::exit((count & 0xFF) as i32);
}
